staging/src/k8s.io/apiserver/pkg/storage/testing/store_tests.go

```diff
@@ -1162,27 +1191,11 @@ func RunTestList(ctx context.Context, t *testing.T, store storage.Interface, com
 			rvMatch:     metav1.ResourceVersionMatchNotOlderThan,
 			expectedOut: []example.Pod{},
 		},
-		{
-			name:        "test consistent List",
-			prefix:      "/pods/empty",
-			pred:        storage.Everything,
-			rv:          "",
-			expectRV:    currentRV,
-			expectedOut: []example.Pod{},
-		},
 	}
 
 	for _, tt := range tests {
 		tt := tt
 		t.Run(tt.name, func(t *testing.T) {
-			// For some asynchronous implementations of storage interface (in particular watchcache),
-			// certain requests may impact result of further requests. As an example, if we first
-			// ensure that watchcache is synchronized up to ResourceVersion X (using Get/List requests
-			// with NotOlderThan semantic), the further requests (even specifying earlier resource
-			// version) will also return the result synchronized to at least ResourceVersion X.
-			// By parallelizing test cases we ensure that the order in which test cases are defined
-			// doesn't automatically preclude some scenarios from happening.
-			t.Parallel()
```

Comment:
I'm not a fan of it - we added it very consciously and it allowed us to catch at least some bugs.
So I really don't want to give up on it...

Reply:
Restored